typos.toml

```diff
@@ -0,0 +1,62 @@
+[files]
+extend-exclude = [
+    # exclude git (sub)modules and generated content
+    "compiler/rustc_codegen_gcc",
+    "compiler/rustc_codegen_cranelift",
+    "compiler/rustc_baked_icu_data",
+    "library/backtrace",
+    "library/stdarch",
+    # generated lorem ipsum texts
+    "library/alloc/benches/str.rs",
+    "library/alloc/tests/str.rs",
+]
+
+[default.extend-words]
+# Add exclusions here, lines should be like `x = "x"`, where `x` is excluded word.
+#
+# Also see docs: https://github.com/crate-ci/typos/blob/v1.28.2/docs/reference.md
+thir = "thir"
+padd = "padd"
+rplace = "rplace"
+arange = "arange"
+rela = "rela"
+unstalled = "unstalled"
+taits = "taits"
+Datas = "Datas"
+splitted = "splitted"
+leafs = "leafs"
+Lits = "Lits"
+makro = "makro"
+optin = "optin"
+parm = "parm"
+unparseable = "unparseable"
+matcheable = "matcheable"
+smove = "smove"
+childs = "childs"
+filetimes = "filetimes"
+
+[default.extend-identifiers]
```

Comment:
Feels like we need comments explaining these exceptions.

Reply:
Added more words for description